components/console-backend-service/internal/domain/k8s/namespace_resolver.go

```diff
@@ -75,6 +60,20 @@ func (r *namespaceResolver) NamespacesQuery(ctx context.Context, applicationName
 		glog.Error(errors.Wrapf(err, "while listing %s", pretty.Namespaces))
 		return nil, gqlerror.New(err, pretty.Namespaces)
 	}
+
+	var namespaces []types.NamespaceWithAdditionalData
+	for _, ns := range rawNamespaces {
+		isSystem := isSystemNamespace(*ns, r.systemNamespaces)
+		passedSystemNamespaceCheck := !isSystem || (withSystemNamespaces != nil && *withSystemNamespaces && isSystem)
```

Comment:
Can be a little more clear IMO:
```suggestion
		passedSystemNamespaceCheck := (withSystemNamespaces != nil && *withSystemNamespaces) || !isSystem
```